saga-spring/src/main/java/io/servicecomb/saga/spring/SagaController.java

```diff
@@ -91,11 +103,104 @@ public SagaController(SagaExecutionComponent sagaExecutionComponent, SagaEventRe
     return ResponseEntity.ok(events);
   }
 
+  @ApiResponses({
+      @ApiResponse(code = 200, response = String.class, message = "success"),
+      @ApiResponse(code = 400, response = String.class, message = "illegal request content"),
+  })
+  @RequestMapping(value = "requests", method = GET)
+  public ResponseEntity<SagaRequestQueryResult> queryRequests(
+      @RequestParam(name = "pageIndex") String pageIndex,
+      @RequestParam(name = "pageSize") String pageSize,
+      @RequestParam(name = "startTime") String startTime,
+      @RequestParam(name = "endTime") String endTime) {
+    if (isRequestParamValid(pageIndex, pageSize, startTime, endTime)) {
+      List<SagaRequest> requests = new ArrayList<>();
+      Page<SagaEventEntity> startEvents = repo.findByTypeAndCreationTimeBetweenOrderByIdDesc(
+          SagaStartedEvent.class.getSimpleName(),
+          new Date(Long.parseLong(startTime)), new Date(Long.parseLong(endTime)),
+          new PageRequest(Integer.parseInt(pageIndex), Integer.parseInt(pageSize)));
+      for (SagaEventEntity event : startEvents) {
```

Comment:
we may want to create a jira for query optimization, in order to avoid keep calling database in a loop

Reply:
OK ,we can improve it later